File: ios-client/StripeApplication/StripeApplication/Data/Models/UpgradeToRecipientResponse.swift
import Foundation

struct UpgradeToRecipientResponse: Codable {
    let id: String
    let stripeAccountId: String
    let isMerchant: Bool
    let isRecipient: Bool
    let merchantCapabilities: [String: AnyCodable]?
    let recipientCapabilities: [String: AnyCodable]?

    enum CodingKeys: String, CodingKey {
        case id
        case stripeAccountId = "stripe_account_id"
        case isMerchant = "is_merchant"
        case isRecipient = "is_recipient"
        case merchantCapabilities = "merchant_capabilities"
        case recipientCapabilities = "recipient_capabilities"
    }
}

File: android-client/app/src/main/java/com/example/stripedemo/data/models/UpgradeToRecipientResponse.kt
package com.example.stripedemo.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class UpgradeToRecipientResponse(
    val id: String,
    @Json(name = "stripe_account_id")
    val stripeAccountId: String,
    @Json(name = "is_merchant")
    val isMerchant: Boolean,
    @Json(name = "is_recipient")
    val isRecipient: Boolean,
    @Json(name = "merchant_capabilities")
    val merchantCapabilities: Map<String, Any?>?,
    @Json(name = "recipient_capabilities")
    val recipientCapabilities: Map<String, Any?>?
)
